backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/floatingnotification/BpkFloatingNotification.kt

```diff
@@ -0,0 +1,202 @@
+  val requiredSize = if (LocalConfiguration.current.screenWidthDp < SMALL_MOBILE_MAX_WIDTH) {
+    RequiredSize(width = SMALL_MOBILE_REQUIRED_WIDTH, height = SMALL_MOBILE_REQUIRED_HEIGHT)
+  } else if (LocalConfiguration.current.screenWidthDp in SMALL_MOBILE_MAX_WIDTH..MOBILE_MAX_WIDTH) {
+    RequiredSize(width = MOBILE_REQUIRED_WIDTH, height = MOBILE_REQUIRED_HEIGHT)
+  } else {
+    RequiredSize(width = TABLET_REQUIRED_WIDTH, height = TABLET_REQUIRED_HEIGHT)
+  }
+
+  AnimatedVisibility(
+    modifier = modifier,
+    visible = hostState.visible,
+    enter = slideInVertically(slideAnimationSpec, initialOffsetY = { it / 2 }) + fadeIn(animationSpec = fadeAnimationSpec),
+    exit = slideOutVertically(slideAnimationSpec, targetOffsetY = { it / 2 }) + fadeOut(animationSpec = fadeAnimationSpec)
+  ) {
+    val data = hostState.data ?: return@AnimatedVisibility
+    Box(
+      modifier = Modifier
+        .fillMaxWidth()
+        .padding(start = BpkSpacing.Base, end = BpkSpacing.Base, bottom = 30.dp),
+      contentAlignment = Alignment.BottomCenter
+    ) {
+      Box(
+        modifier = Modifier
+          .clip(RoundedCornerShape(BpkBorderRadius.Md))
+          .background(color = BpkTheme.colors.corePrimary)
+          .requiredSize(height = requiredSize.height, width = requiredSize.width)
+          .padding(start = BpkSpacing.Base, end = BpkSpacing.Base),
+        contentAlignment = Alignment.CenterStart
+      ) {
+        Row(
+          horizontalArrangement = Arrangement.spacedBy(BpkSpacing.Base),
+          verticalAlignment = Alignment.CenterVertically
+        ) {
+          data.icon?.let { icon ->
+            BpkIcon(
+              icon = icon,
+              contentDescription = null,
+              size = BpkIconSize.Small,
+              tint = BpkTheme.colors.textOnDark
+            )
+          }
+          BpkText(
+            modifier = Modifier.weight(0.8f),
+            text = data.text,
+            maxLines = 2,
+            color = BpkTheme.colors.textOnDark,
+            style = BpkTheme.typography.footnote
+          )
+          data.cta?.let { cta ->
+            Box(
+              modifier = Modifier
+                .weight(0.2f)
+                .sizeIn(minHeight = BpkSpacing.Xxl, minWidth = BpkSpacing.Xxl)
+                .clickable { cta.onClick.invoke() },
+              contentAlignment = Alignment.Center
+            ) {
+              BpkText(
+                text = cta.text,
+                textAlign = TextAlign.Center,
+                color = BpkTheme.colors.textOnDark,
+                style = BpkTheme.typography.label2
+              )
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+data class Cta(
+  val text: String,
+  val onClick: () -> Unit,
+)
+
+internal data class BpkFloatingNotificationData(
+  val text: String,
+  val icon: BpkIcon? = null,
+  val cta: Cta? = null,
+)
+
+@Stable
+class BpkFloatingNotificationState(initiallyVisible: Boolean) {
```

Comment:
We need to follow up with the stack Snackbar designs, otherwise, the developers will have to create a state for each possible snack bar reason.

Reply:
no we don't need to and neither do developers. I've tested it, we can have one state and use `show()` method several times with different texts and icons.